File: src/framework/controllers/friendRequest/deny.ts
import {
  AuthenticatedLocals,
  IRequest,
  IResponse,
} from "../../../common/types";
import { z } from "zod";
import { Controller, HttpMethod } from "../types";
import { authMiddleware } from "../../middleware";
import { databases().friendRequestDatabase } from "../../../domain/repositories";

export class DenyFriendRequestController implements Controller<void> {
  method = HttpMethod.POST;

  route = "/friendRequest/deny";

  middleware = [authMiddleware];

  validator = z.object({
    senderCognitoId: z.string(),
  });

  execute(
    req: IRequest<z.infer<typeof this.validator>>,
    res: IResponse<void, AuthenticatedLocals>
  ) {
    return databases().friendRequestDatabase.deny(
      req.body.senderCognitoId,
      res.locals.cognitoId
    );
  }
}

File: lib/src/data/backend/friendRequest/deny.dart
part of backend;

extension Deny on FriendRequest {
  Future<Failure?> deny({
    required String accessToken,
    required String senderCognitoId,
  }) =>
      _request(
        route: "/friendRequest/deny",
        method: HttpMethod.post,
        authorization: accessToken,
        data: {
          'senderCognitoId': senderCognitoId,
        },
      );
}